### DISCUSSION TITLE
Speedtest Server Meta Data API (needs feedback)

### DISCUSSION THREAD

**Discussion by alexjustesen** [alexjustesen/speedtest-tracker#245] [Ideas]
### Problem
Currently when searching for an Ookla Speedtest Server (feature introduced in v0.9.0 in PR #216) there is no way to search a complete list of servers or return the server's meta data (server name, url/ip address, ect.) after it's been selected.

### Proposed Solution
Create a database and API to both collect and list server meta data that can be accessed from a central location like https://api.speedtest-tracker.dev.

### Problems to Consider
1. Collect known servers with their meta data and add the data to the database. I'll need a solution which is both efficient and privacy focused.
2. Periodically validate each server in the database to make sure they're online and functioning.
3. Are there any legal considerations, like can I even collect this data?
4. tbd...

### Resources
- https://gist.github.com/ofou/654efe67e173a6bff5c64ba26c09d058
- https://sparanoid.com/lab/speedtest-list/

**Comment by alexdelprete** [comment]
This is a great idea/feature. Let me know how I can help you.
